### ISSUE TITLE
Help function for creating tensor list

### ISSUE THREAD

**Issue by AlexandreEichenberger** [onnx/onnx-mlir#644]
I created a function that build a tensor list based on the signature returned by `omInputSignature()`

It is pretty simple and use a rudimentary JSON reader lifted from GitHub with a MIT license (very permissive).

Its interface is very simple:

```
OMTensorList *omTensorListCreateFromInputSignature(bool trace = false);
```

and assets if there are issues (could also just return null).

My question is
1. Would you find that useful? To me it is to generate a simple program that calls in any models.
2. If useful, should it be added to OMTensor? Or OMTensorHelper? Interface is in C.

And if we want it, how do we integrate the JSON reader? Its about 300 LOC, one .h and .cpp.

**Comment by AlexandreEichenberger** [comment]
Right now, it creates the data blocks and leave uninitialized. One could add another parameter to determine if initializing, or provide a list of blocks and setting each corresponding block to the corresponding element in the list.

**Comment by AlexandreEichenberger** [comment]
Possible interface (currently implemented)

``` c++
/**
 * \brief Create and initialize an OMTensorList from the signature of a model
 *
 * This function parse the signature of a ONNX compiled network, attached to the
 * binary via a .so and will scan the JSON signature for its input. For each
 * input in turn, it create a tensor of the proper type and shape. Data will be
 * either initialized (if dataPtrList is provided), allocated (if dataPtrList is
 * null and dataAlloc is set to true), or will otherwise be left empty. In case
 * of errors, a null pointer returned.
 *
 *
 * @param dataPtrList Pointer to a list of data pointers of the right size, as
 * determined by the signature.
 * @param allocData When no dataPtrList is provided, the this boolean variable
 * determine if data is to be allocated or not, using the sizes determined by
 * the signature.
 * @param trace If true, provide a printout of the signatures (input and
 * putput).
 * @return pointer to the TensorList just created, or null on error.
 */
OMTensorList *omTensorListCreateFromInputSignature(
    void **dataPtrList, bool dataAlloc, bool trace);
```

**Comment by tjingrant** [comment]
I think this makes sense to me per se, although we probably want to be careful about the impact of including a json reader; including it will likely entail building it during onnx-mlir installation, and linking with it (along with other stuffs like our runtime library) at the end of model compilation each time. I'd also be interested to learn about the motivations, it feels like it may reveal some design flaws of the current c99 runtime interface.

**Comment by AlexandreEichenberger** [comment]
implemented by #709 which uses LLVM native json reader
